src/tsdf/tsdf_2d.cpp

```diff
@@ -0,0 +1,84 @@
+//  ================================================================
+//  Created by Fei Shan on 03/19/19.
+//  ================================================================
+
+
+
+
+//local
+#include "tsdf.hpp"
+#include "ewa_common.hpp"
+#include "ewa.hpp"
+
+namespace tsdf {
+
+eig::MatrixXf generate_TSDF_2D(
+        int image_y_coordinate,
+        const eig::Matrix<unsigned short, eig::Dynamic, eig::Dynamic> &depth_image,
+        float depth_unit_ratio,
+        const eig::Matrix3f &camera_intrinsic_matrix,
+        const eig::Matrix4f &camera_pose,
+        const eig::Vector3i &array_offset,
+        int field_size,
+        float voxel_size,
+        int narrow_band_width_voxels,
+        float default_value) {
+    eig::MatrixXf field(field_size, field_size);
+    std::fill_n(field.data(), field.size(), default_value);
+    float narrow_band_half_width = static_cast<float>(narrow_band_width_voxels / 2) * voxel_size;
+
+    float w_voxel = 1.0f;
+    float y_voxel = 0.0f;
+
+    int matrix_size = static_cast<int>(field.size());
+
+#pragma omp parallel for
+    for (int i_element = 0; i_element < matrix_size; i_element++) {
+        // Any MatrixXf in Eigen is column-major
+        // i_element = x * column_count + y
+        div_t division_result = std::div(i_element, field_size);
+        int x_field = division_result.quot;
+        int y_field = division_result.rem;
```

Comment:
please revise to not use std::div operation, use "%" and "/" operators instead. Compiler will auto-optimize those to a single operation.

Reply:
fixed